Documentation.md

```diff
@@ -164,590 +174,606 @@ and made available to your JS client code.
+agent.setState(state, {
+   success: function() { /* ... */ },
+   failure: function(err) { /* ... */ }
 });
 ```
-Set the agent's current availability state.  Can only be performed if the agent is not handling a live contact.
+Set the agent's current availability state. Can only be performed if the agent is not handling a live contact.
 
-You can optionally provide success and failure callbacks to determine whether the
-operation succeeded.
+Optional success and failure callbacks can be provided to determine if the operation was successful.
 
 ### `agent.connect()`
-```
-agent.connect(endpoint, {
-   queueARN: QUEUE_ARN,
-   success: function() { ... },
-   failure: function(err) { ... }
-});
-```
-Creates an outbound contact to the given endpoint.  You can optionally provide a queueARN to associate the contact with a queue.
-
-You can optionally provide success and failure callbacks to determine whether the operation succeeded.
-
-Example use to create a new outbound call directly to a phone number.
-```
-var endpoint = connect.Endpoint.byPhoneNumber("+11234567");
+```js
+var endpoint = connect.Endpoint.byPhoneNumber("+18005550100");
 var agent = new connect.Agent();
-var queueArn = "arn:aws:connect:us-east-1:12345678:instance/<CONNECT_INSTANCE_ID>/queue/<CONNECT_QUEUE_ID>";
+var queueArn = "arn:aws:connect:<REGION>:<ACCOUNT_ID>:instance/<CONNECT_INSTANCE_ID>/queue/<CONNECT_QUEUE_ID>";
 
 agent.connect(endpoint, {
   queueARN: queueArn,
-  success: function() { console.log("outbound call connected") },
+  success: function() { console.log("outbound call connected"); },
   failure: function(err) {
     console.log("outbound call connection failed");
     console.log(err);
   }
 });
 ```
+Creates an outbound contact to the given endpoint. You can optionally provide a `queueARN` to associate the contact with a queue.
 
-### `agent.getAllQueueARNs`
-```
-const ARNs = agent.getAllQueueARNs;
-```
-Returns a list of the ARNs associated with this agent's routing profile's queues. 
+Optional success and failure callbacks can be provided to determine if the operation was successful.
 
-### `agent.getEndpoints` / `agent.getAddresses`
+### `agent.getAllQueueARNs()`
+```js
+var ARNs = agent.getAllQueueARNs();
 ```
-const endpoints = agent.getEndpoints(
-   queue_array,
+Returns a list of the ARNs associated with this agent's routing profile's queues.
+
+### `agent.getEndpoints()` / `agent.getAddresses()`
+```js
+var queuesARNs = agent.getAllQueueARNs();
+var endpoint1 = connect.Endpoint.byPhoneNumber("+18005550101");
+var endpoint2 = connect.Endpoint.byPhoneNumber("+18005550102");
+agent.getEndpoints(
+   queuesARNs,
    {
-      succes: function_1,
-      failure: function_2
+      success: function(data) {
+         var endpoints = data.endpoints; // or data.addresses
+      },
+      failure: function(err) {
+      }
    },
    {
-      endpoints: [endpoint1, endpoint2, etc.]
+      endpoints: [endpoint, endpoint2]
       maxResults: 95
-); 
+   }
+);
```

Comment:
Found out `agent.getEndpoints()` is an async callback-based function (i.e. it does not return a value).